src/main/java/com/github/manolo8/darkbot/core/objects/facades/QuestProxy.java

```diff
@@ -0,0 +1,323 @@
+package com.github.manolo8.darkbot.core.objects.facades;
+
+import com.github.manolo8.darkbot.core.itf.Updatable;
+import com.github.manolo8.darkbot.core.objects.swf.FlashList;
+import com.github.manolo8.darkbot.core.objects.swf.ObjArray;
+
+import eu.darkbot.api.managers.QuestAPI;
+
+import static com.github.manolo8.darkbot.Main.API;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jetbrains.annotations.Nullable;
+
+public class QuestProxy extends Updatable implements QuestAPI {
+    private @Nullable QuestProxy.Quest currentQuest;
+
+    private final List<QuestProxy.QuestListItem> questItems = new ArrayList<>();
+
+    private final ObjArray questItemsArr = ObjArray.ofArrObj(true);
+
+    private @Nullable QuestProxy.Quest questGiverSelected;
+    private @Nullable QuestProxy.QuestListItem questInfoGiverSelected;
+
+    private boolean visibleQuestGiver;
+    private int tabSelected;
+
+    @Override
+    public void update() {
+        if (address == 0) {
+            return;
+        }
+
+        long questClass = API.readMemoryPtr(address + 0x98);
+
+        long currentQuestAddr = API.readMemoryPtr(questClass + 0x28);
+        if (this.currentQuest == null) {
+            this.currentQuest = new Quest();
```

Comment:
we generally avoid having null objects and needing to initialize them, consider just having the `currentQuest = new Quest()` be the default value, and in getCurrentQuest just do `quest.address == 0 ? null : quest;`, same for questInfoGiverSelected